hail/python/hailtop/aiocloud/aioazure/fs.py

```diff
@@ -370,7 +370,14 @@ def get_url_parts(url: str) -> Tuple[str, str, str, str]:
             assert name[0] == '/'
             name = name[1:]
 
-        token = match.groupdict()['token']
+        token = None
+        # Look for a terminating SAS token.
+        if (query_index := name.rfind('?')) != -1:
+            query_string = name[query_index + 1:]
+            # We will accept it as a token string if it has at least 1 key-value pair of the form 'k=v'.
```

Comment:
"We will accept it as a token string if it begins with at least 1 key-value pair of the form 'k=v'."